### ISSUE TITLE
Packing fails with : `No such file or directory: 'C:\\Python36\\Scripts\\pyinstaller.exe\\__main__.pyo'`

### ISSUE THREAD

**Issue by Enteee** [pyinstaller/pyinstaller#4419]
When building an onefile executable which contains `pillow` and `opencv` I get a `FileNotFoundError: [Errno 2] No such file or directory: 'C:\\Python36\\Scripts\\pyinstaller.exe\\__main__.pyo'` - error .

# Reproduce

For simplicity I tried to create a minimal project [here](https://github.com/Enteee/pyinstaller-bug-packaging) which can be used to reproduce the issue.

requires:
*  linux
* docker
```
$ git clone https://github.com/Enteee/pyinstaller-bug-packaging.git
$ cd pyinstaller-bug-packaging
$ ./build.sh --windows
```

What this will do:
1. Downloads the [`cdrx/pyinstaller-windows`](https://hub.docker.com/r/cdrx/pyinstaller-windows) image. This contains a `pyinstaller` 3.5 installation as well as `wine`.
2. Installs all packages from [`requirements.txt`](https://github.com/Enteee/pyinstaller-bug-packaging/blob/master/requirements.txt)
3. Runs `pyinstaller` with `wine`  using [minimal.spec](https://github.com/Enteee/pyinstaller-bug-packaging/blob/master/minimal.spec)
4. Eventually yields: `FileNotFoundError: [Errno 2] No such file or directory: 'C:\\Python36\\Scripts\\pyinstaller.exe\\__main__.pyo'`

<details><summary>SHOW FULL OUTPUT</summary>
<p>

```
Sending build context to Docker daemon  2.048kB
Step 1/3 : ARG from_image_name
Step 2/3 : ARG from_image_tag
Step 3/3 : FROM "${from_image_name}:${from_image_tag}"
 ---> ac1227af8417
Successfully built ac1227af8417
Successfully tagged build-test-windows:python3
Collecting qrcode (from -r requirements.txt (line 1))
  Downloading https://files.pythonhosted.org/packages/42/87/4a3a77e59ab7493d64da1f69bf1c2e899a4cf81e51b2baa855e8cc8115be/qrcode-6.1-py2.py3-none-any.whl
Collecting pillow (from -r requirements.txt (line 2))
  Downloading https://files.pythonhosted.org/packages/4e/d9/468422371e6fcf02d6a162ee30db4552221de8b2b3ff837363bf54cbb347/Pillow-6.1.0-cp36-cp36m-win_amd64.whl (2.0MB)
Collecting opencv-python-headless (from -r requirements.txt (line 3))
  Downloading https://files.pythonhosted.org/packages/f3/c7/0b33e00a4ed0f34444cab076bb61edf0b70c6a56c1b4a1cd241d388fe34e/opencv_python_headless-4.1.1.26-cp36-cp36m-win_amd64.whl (39.0MB)
Collecting six (from qrcode->-r requirements.txt (line 1))
  Downloading https://files.pythonhosted.org/packages/73/fb/00a976f728d0d1fecfe898238ce23f502a721c0ac0ecfedb80e0d88c64e9/six-1.12.0-py2.py3-none-any.whl
Collecting colorama; platform_system == "Windows" (from qrcode->-r requirements.txt (line 1))
  Downloading https://files.pythonhosted.org/packages/4f/a6/728666f39bfff1719fc94c481890b2106837da9318031f71a8424b662e12/colorama-0.4.1-py2.py3-none-any.whl
Collecting numpy>=1.11.3 (from opencv-python-headless->-r requirements.txt (line 3))
  Downloading https://files.pythonhosted.org/packages/87/f4/682e88f2b5c1d49a8011fadee57eb3c13f55f156536597a625109261314d/numpy-1.17.1-cp36-cp36m-win_amd64.whl (12.8MB)
Installing collected packages: six, colorama, qrcode, pillow, numpy, opencv-python-headless
Successfully installed colorama-0.4.1 numpy-1.17.1 opencv-python-headless-4.1.1.26 pillow-6.1.0 qrcode-6.1 six-1.12.0

115 INFO: PyInstaller: 3.5
115 INFO: Python: 3.6.4
118 INFO: Platform: Windows-XP-5.1.2600
124 INFO: UPX is not available.
124 INFO: Removing temporary files and cleaning cache in C:\users\root\Application Data\pyinstaller
126 INFO: Extending PYTHONPATH with paths
['Z:\\src', 'Z:\\src']
126 INFO: checking Analysis
126 INFO: Building Analysis because Analysis-00.toc is non existent
126 INFO: Initializing module dependency graph...
131 INFO: Initializing module graph hooks...
133 INFO: Analyzing base_library.zip ...
3673 INFO: running Analysis Analysis-00.toc
3685 INFO: Adding Microsoft.Windows.Common-Controls to dependent assemblies of final executable
  required by c:\Python36\python.exe
4088 INFO: Caching module hooks...
4098 INFO: Analyzing minimal.py
4189 INFO: Processing pre-safe import module hook   six.moves
4830 INFO: Processing pre-find module path hook   distutils
6549 INFO: Processing pre-find module path hook   site
6550 INFO: site: retargeting to fake-dir 'c:\\Python36\\lib\\site-packages\\PyInstaller\\fake-modules'
8068 INFO: Processing pre-safe import module hook   setuptools.extern.six.moves
8601 INFO: Processing pre-safe import module hook   win32com
Traceback (most recent call last):
  File "<string>", line 2, in <module>
ModuleNotFoundError: No module named 'win32com'
11286 INFO: Loading module hooks...
11286 INFO: Loading module hook "hook-distutils.py"...
11287 INFO: Loading module hook "hook-encodings.py"...
11438 INFO: Loading module hook "hook-lib2to3.py"...
11448 INFO: Loading module hook "hook-numpy.core.py"...
11639 INFO: Loading module hook "hook-numpy.py"...
11641 INFO: Loading module hook "hook-PIL.Image.py"...
12213 INFO: Loading module hook "hook-PIL.py"...
12215 INFO: Excluding import 'PyQt5'
12217 INFO:   Removing import of PyQt5 from module PIL.ImageQt
12218 INFO: Import to be excluded not found: 'FixTk'
12218 INFO: Excluding import 'PyQt4'
12221 INFO:   Removing import of PyQt4 from module PIL.ImageQt
12222 INFO: Excluding import 'PySide'
12224 INFO:   Removing import of PySide from module PIL.ImageQt
12225 INFO: Excluding import 'tkinter'
12229 INFO:   Removing import of tkinter from module PIL.ImageTk
12229 INFO: Loading module hook "hook-PIL.SpiderImagePlugin.py"...
12231 INFO: Excluding import 'tkinter'
12232 INFO: Import to be excluded not found: 'FixTk'
12233 INFO: Loading module hook "hook-pkg_resources.py"...
12837 INFO: Processing pre-safe import module hook   win32com
Traceback (most recent call last):
  File "<string>", line 2, in <module>
ModuleNotFoundError: No module named 'win32com'
12929 INFO: Processing pre-safe import module hook   win32com
Traceback (most recent call last):
  File "<string>", line 2, in <module>
ModuleNotFoundError: No module named 'win32com'
13018 INFO: Loading module hook "hook-pydoc.py"...
13019 INFO: Loading module hook "hook-setuptools.py"...
13083 INFO: Loading module hook "hook-sysconfig.py"...
13085 INFO: Loading module hook "hook-xml.dom.domreg.py"...
13086 INFO: Loading module hook "hook-xml.py"...
13086 INFO: Loading module hook "hook-_tkinter.py"...
13396 INFO: checking Tree
13396 INFO: Building Tree because Tree-00.toc is non existent
13396 INFO: Building Tree Tree-00.toc
13451 INFO: checking Tree
13451 INFO: Building Tree because Tree-01.toc is non existent
13451 INFO: Building Tree Tree-01.toc
13500 INFO: Looking for ctypes DLLs
13547 INFO: Analyzing run-time hooks ...
13554 INFO: Including run-time hook 'pyi_rth_multiprocessing.py'
13559 INFO: Including run-time hook 'pyi_rth_pkgres.py'
13574 INFO: Looking for dynamic libraries
14053 INFO: Looking for eggs
14053 INFO: Using Python library c:\Python36\python36.dll
14053 INFO: Found binding redirects: 
[]
Traceback (most recent call last):
  File "c:\Python36\lib\runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "c:\Python36\lib\runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "C:\Python36\Scripts\pyinstaller.exe\__main__.py", line 9, in <module>
  File "c:\Python36\lib\site-packages\PyInstaller\__main__.py", line 111, in run
    run_build(pyi_config, spec_file, **vars(args))
  File "c:\Python36\lib\site-packages\PyInstaller\__main__.py", line 63, in run_build
    PyInstaller.building.build_main.main(pyi_config, spec_file, **kwargs)
  File "c:\Python36\lib\site-packages\PyInstaller\building\build_main.py", line 844, in main
    build(specfile, kw.get('distpath'), kw.get('workpath'), kw.get('clean_build'))
  File "c:\Python36\lib\site-packages\PyInstaller\building\build_main.py", line 791, in build
    exec(code, spec_namespace)
  File "minimal.spec", line 17, in <module>
    noarchive=True)
  File "c:\Python36\lib\site-packages\PyInstaller\building\build_main.py", line 243, in __init__
    self.__postinit__()
  File "c:\Python36\lib\site-packages\PyInstaller\building\datastruct.py", line 158, in __postinit__
    self.assemble()
  File "c:\Python36\lib\site-packages\PyInstaller\building\build_main.py", line 597, in assemble
    for name, path, typecode in compile_py_files(new_toc, CONF['workpath']):
  File "c:\Python36\lib\site-packages\PyInstaller\utils\misc.py", line 150, in compile_py_files
    with open(obj_fnm, 'rb') as fh:
FileNotFoundError: [Errno 2] No such file or directory: 'C:\\Python36\\Scripts\\pyinstaller.exe\\__main__.pyo'
```

</p>
</details>

I am not sure what is going on, by what I noticed is that when I change `noarchive` to `False` the build works just fine. This is most likely because it then skips the archive building which fails with the error.

<details><summary>SHOW DIFF FOR THE CHANGE</summary>
<p>

```diff
diff --git a/minimal.spec b/minimal.spec
index 2a0a10c..1c9da82 100644
--- a/minimal.spec
+++ b/minimal.spec
@@ -14,7 +14,7 @@ a = Analysis(['minimal.py'],
              win_no_prefer_redirects=False,
              win_private_assemblies=False,
              cipher=block_cipher,
-             noarchive=True)
+             noarchive=False)
 pyz = PYZ(a.pure, a.zipped_data,
              cipher=block_cipher)
 exe = EXE(pyz,
```

</p>
</details>

## Possibly related issues

* #3977

**Comment by stevenliu1970** [comment]
same problem to me, any one can help? thanks

**Comment by rokm** [comment]
The provided reproducer does not work anymore - it now uses python 3.7 and pyinstaller 3.6, and finishes without the error.

However, #3995 also has a reproducer, which shows that the issue was present in pyinstaller 3.5 but has been fixed in 3.6. According to git bisect, the fixing commit was c84ddebbd5bc96dc1e1de57fa60994734242edcf.

Based on that, I'm closing this issue.
